PathfinderAPI/Replacements/ExtensionInfoLoader.cs

```diff
@@ -0,0 +1,176 @@
+        {
+            EventManager.onPluginUnload += OnPluginUnload;
+        }
+        
+        private static void OnPluginUnload(Assembly pluginAsm)
+        {
+            var allTypes = AccessTools.GetTypesFromAssembly(pluginAsm);
+            CustomExecutors.RemoveAll(x => allTypes.Contains(x.ExecutorType));
+        }
+        
+        [HarmonyPrefix]
+        [HarmonyPatch(typeof(ExtensionInfo), nameof(ExtensionInfo.ReadExtensionInfo))]
+        private static bool LoadExtensionInfoPrefix(string folderpath, ref ExtensionInfo __result)
+        {
+            __result = LoadExtensionInfo(folderpath);
+            return false;
+        }
+
+        public static ExtensionInfo LoadExtensionInfo(string folderpath)
+        {
+            var filepath = folderpath + "/ExtensionInfo.xml";
+
+            if (!File.Exists(filepath))
+                throw new FileNotFoundException($"Could not find ExtensionInfo.xml in folder {folderpath}");
+
+            var extInfo = new ExtensionInfo();
+            extInfo.FolderPath = folderpath;
+            extInfo.Language = "en-us";
+            
+            var logopath = folderpath + "/Logo.";
+            string extension = null;
+
+            if (File.Exists(logopath + "png"))
+                extension = "png";
+            else if (File.Exists(logopath + "jpg"))
+                extension = "jpg";
+
+            if (extension != null)
+                using (var input = File.OpenRead(logopath + extension))
+                    extInfo.LogoImage = Texture2D.FromStream(Game1.getSingleton().GraphicsDevice, input);
+
+            var exe = new EventExecutor(LocalizedFileLoader.GetLocalizedFilepath(filepath), true);
+
+            foreach (var custom in CustomExecutors)
+            {
+                var instance = (ExtensionInfoExecutor)Activator.CreateInstance(custom.ExecutorType);
+                instance.Init(ref extInfo);
+                exe.RegisterExecutor(custom.Element, instance.Execute, custom.Options);
+            }
+
+            exe.Reg("Name", (exec, info) => extInfo.Name = Utils.CleanStringToLanguageRenderable(info.Content));
+            exe.Reg("Language", (exec, info) =>
+            {
+                var language = info.Content;
+                if (!ValidLanguages.Contains(language))
+                    throw new FormatException($"Invalid language in '{filepath}': {language}");
+                extInfo.Language = language;
+            });
+            exe.Reg("AllowSaves", (exec, info) => extInfo.AllowSave = info.ContentAsBoolean(filepath));
+            exe.Reg("StartingVisibleNodes", (exec, info) =>
+                extInfo.StartingVisibleNodes = info.Content
+                    .Replace(@"\s+", "")
+                    .Split(','));
+            exe.Reg("StartingMission", (exec, info) => extInfo.StartingMissionPath = info.Content);
+            exe.Reg("StartingActions", (exec, info) => extInfo.StartingActionsPath = info.Content);
```

Comment:
Both StartingMission and StartingActions should evaluate as null when `NONE` is passed in, see the related base game code:
```cs
				if (xmlReader.Name == "StartingMission")
				{
					xmlReader.MoveToContent();
					extensionInfo.StartingMissionPath = xmlReader.ReadElementContentAsString();
					if (extensionInfo.StartingMissionPath == "NONE")
					{
						extensionInfo.StartingMissionPath = null;
					}
				}
				if (xmlReader.Name == "StartingActions")
				{
					xmlReader.MoveToContent();
					extensionInfo.StartingActionsPath = xmlReader.ReadElementContentAsString();
					if (extensionInfo.StartingActionsPath == "NONE")
					{
						extensionInfo.StartingActionsPath = null;
					}
				}
```